### DISCUSSION TITLE
Switch ApacheConf to a maintained upstream

### DISCUSSION THREAD

**Discussion by nisbet-hubbard** [github-linguist/linguist#7135] [General]
The upstream project for ApacheConf hasn’t been maintained for the past nine years, and is unlikely to see more work done because the product it was originally created for was also abandoned: https://github.com/textmate/apache.tmbundle

The ApacheConf grammar, however, is still very much incomplete. Shall we switch to a different upstream such as [apache.vim](https://github.com/vim/vim/blob/master/runtime/syntax/apache.vim)?

If not, someone would need to fork textmate/apache.tmbundle so it can start accepting PRs again.

**Comment by lildude** [comment]
> The ApacheConf grammar, however, is still very much incomplete. Shall we switch to a different upstream such as [apache.vim](https://github.com/vim/vim/blob/master/runtime/syntax/apache.vim)?

Nope. Why? Because we only support TextMate compatible grammars. 

> If not, someone would need to fork textmate/apache.tmbundle so it can start accepting PRs again.

Yup, or find another actively maintained fork or VSCode version.

**Reply by nisbet-hubbard** [reply]
Thanks for the pointers!

I’ve updated Apache.plist for Apache 2.4.62/2.5 and tested it on TextMate. It’s now ready to be pulled: https://github.com/nisbet-hubbard/apache.tmbundle.

I’ll make sure this fork gets regularly maintained.

**Comment by nisbet-hubbard** [comment]
After comparing the VSCode version, I found it to be a better basis for future updates. Since the maintainer of that version is still active, in #7143 I’ve changed the source to his version and will contribute to that repo.
